common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -233,4 +233,49 @@ public static String getErrorMessage(String errorMessage, String modelId, Boolea
             return errorMessage + " Model ID: " + modelId;
         }
     }
+
+    /**
+     * Collects the prefixes of the toString() method calls present in the values of the given map.
+     *
+     * @param map A map containing key-value pairs where the values may contain toString() method calls.
+     * @return A list of prefixes for the toString() method calls found in the map values.
+     */
+    public static List<String> collectToStringPrefixes(Map<String, String> map) {
+        List<String> prefixes = new ArrayList<>();
+        for (String key : map.keySet()) {
+            String value = map.get(key);
+            if (value != null) {
+                Pattern pattern = Pattern.compile("\\$\\{(\\w+\\.\\w+)\\.toString\\(\\)\\}");
```

Comment:
For connector, we should only check `${parameters.xxx}` , but this pattern will also check `${model_config.xxx}`.

Reply:
changed to regex pattern to `"\\$\\{parameters\\.(.+?)\\.toString\\(\\)\\}"`